src/auth/handlers.rs

```diff
@@ -46,24 +44,28 @@ async fn register(
 async fn login(
     ctx: Data<Context>,
     Json(creds): Json<SignInCredentials>,
-) -> Result<HttpResponse> {
+) -> Result<HttpResponse, APIError> {
     log::trace!("Received login request");
 
-    let Ok(user) = ctx.db.get_user_by_creds(&creds).await else {
-        let _ = bcrypt::hash(&creds.password, bcrypt::DEFAULT_COST)?;
-        return Ok(HttpResponse::Unauthorized().finish());
-    };
+    let user = ctx.db.get_user_by_creds(&creds).await.map_err(|_| {
+        // NOTE(t3m8ch): This line will never throw an error because it depends
```

Comment:
Rust никогда не "бросает" ничего. Возможно, будет лучше:
> This call will never end with an error because...

Reply:
И я из комментария не понял, почему пошли на `unwrap()`.
> This call will never end with an error, because it can only produce one, when the cost is invalid, which we cannot possibly have.